GCCcore-6.4.0

Dear Easy-Builders,

I try to build the quite old compiler gcc-6.4.0 using

> eb GCCcore-6.4.0.eb

and that build fails. But the logfile, which is appended to the issue, contains only the first 300 characters, which show no error at all. Therefore I went to the corresponding build directory, run make again, and get the compiler error:

> ../../../../libsanitizer/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc: In function ‘int __sanitizer::TracerThread(void*)’:
../../../../libsanitizer/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc:270:22: error: aggregate ‘sigaltstack handler_stack’ has incomplete type and cannot be defined
   struct sigaltstack handler_stack;

What puzzles me, is that there is already a [patch for that problem](https://gist.github.com/froland/322a3320fcabd46a5702bd4a2a228198) and the patch is there. The logfile also states, that it applies the patch.

Do you have any hints, what I could do to correct that, or what I do wrong?

I see also a small relation to the [issue easybuilders/easybuild-easyblocks#495 on easybuild](https://github.com/easybuilders/easybuild/issues/495), which got a response stating, that gcc 8.2 might lead to problems with binutils 2.30. My system has installed the gcc-7.4.0, which is also newer than the gcc-6.4.0. 

My easybuild-version is:

> eb --version
This is EasyBuild 4.0.1 (framework: 4.0.1, easyblocks: 4.0.1) on host moivre.

[easybuild-nEWZ6N.log](https://github.com/easybuilders/easybuild-easyblocks/files/3870451/easybuild-nEWZ6N.log)

--- a/easybuild/easyconfigs/g/GCCcore/GCCcore-6.4.0.eb
+++ b/easybuild/easyconfigs/g/GCCcore/GCCcore-6.4.0.eb
@@ -3,7 +3,7 @@ easyblock = 'EB_GCC'
 name = 'GCCcore'
 version = '6.4.0'
 
-homepage = 'http://gcc.gnu.org/'
+homepage = 'https://gcc.gnu.org/'
 
 description = """
  The GNU Compiler Collection includes front ends for C, C++, Objective-C,
@@ -16,13 +16,13 @@ toolchain = SYSTEM
 local_mpfr_version = '3.1.5'
 
 source_urls = [
-    'http://ftpmirror.gnu.org/gnu/gcc/gcc-%(version)s',  # GCC auto-resolving HTTP mirror
-    'http://ftpmirror.gnu.org/gnu/gmp',  # idem for GMP
-    'http://ftpmirror.gnu.org/gnu/mpfr',  # idem for MPFR
-    'http://ftpmirror.gnu.org/gnu/mpc',  # idem for MPC
+    'https://ftpmirror.gnu.org/gnu/gcc/gcc-%(version)s',  # GCC auto-resolving HTTP mirror
+    'https://ftpmirror.gnu.org/gnu/gmp',  # idem for GMP
+    'https://ftpmirror.gnu.org/gnu/mpfr',  # idem for MPFR
+    'https://ftpmirror.gnu.org/gnu/mpc',  # idem for MPC
     'ftp://gcc.gnu.org/pub/gcc/infrastructure/',  # GCC dependencies
-    'http://gcc.cybermirror.org/infrastructure/',  # HTTP mirror for GCC dependencies
-    'http://isl.gforge.inria.fr/',  # original HTTP source for ISL
+    'http://gcc.cybermirror.org/infrastructure/',  # HTTP (no HTTPS) mirror for GCC dependencies
+    'https://isl.gforge.inria.fr/',  # original HTTP source for ISL
 ]
 sources = [
     'gcc-%(version)s.tar.gz',

--- a/easybuild/easyconfigs/g/GCCcore/GCCcore-6.4.0.eb
+++ b/easybuild/easyconfigs/g/GCCcore/GCCcore-6.4.0.eb
@@ -35,6 +35,7 @@ patches = [
     ('mpfr-%s-allpatches-20170606.patch' % local_mpfr_version, '../mpfr-%s' % local_mpfr_version),
     'GCCcore-6.2.0-fix-find-isl.patch',
     'GCCcore-%(version)s_fix-linux-unwind-fix-ucontext.patch',
+    'GCCcore-6.3.0_fix-sanitizer_linux.patch'
 ]
 checksums = [
     '4715f02413f8a91d02d967521c084990c99ce1a671b8a450a80fbd4245f4b728',  # gcc-6.4.0.tar.gz
@@ -46,6 +47,7 @@ checksums = [
     '5ad909606d17d851c6ad629b4fddb6c1621844218b8d139fed18c502a7696c68',  # GCCcore-6.2.0-fix-find-isl.patch
     # GCCcore-6.4.0_fix-linux-unwind-fix-ucontext.patch
     '368d027a7c0ef711188445c6b2efe1837150d9658872c6936162e43822e32ae4',
+    '650e903af7399b6dca82bb606560a5d9bfe06f794a73d3ada9018bdc8ab497a3',  # GCCcore-6.3.0_fix-sanitizer_linux.patch
 ]
 
 builddependencies = [

